
focused.csv.gz and focused.T.csv.gz are rounded to 1000.


In [1]:
import numpy as np
print(np.__version__)

1.19.5


In [2]:
import pandas as pd
print(pd.__version__)

1.1.5


In [3]:
import matplotlib
matplotlib.__version__

'3.3.4'

In [4]:
import matplotlib.pyplot as plt

In [5]:
# Adjust plot size
#options(repr.plot.width=16, repr.plot.height=6)

# For some reason, if this is in the same cell as the import command, it doesn't work.
plt.rcParams["figure.figsize"] = [18.0,8.0]

# Read Metadata 1

In [6]:
mt1=pd.read_csv("TCGA/metadata.cart.TCGA.GBM-LGG.WGS.bam.2020-07-17.csv",header=None,
 names=['submitter','days_to_death','gender','disease_type','center_name'])

In [7]:
mt1.shape

(127, 5)

In [8]:
mt1.rename(columns={"submitter": "subject"},inplace=True)
mt1.rename(columns={"gender": "gender1"},inplace=True)

In [9]:
mt1.head()

subject  days_to_death gender1             disease_type  \
0  02-2483            NaN    male  Glioblastoma Multiforme   
1  02-2485            NaN    male  Glioblastoma Multiforme   
2  06-0125         1448.0  female  Glioblastoma Multiforme   
3  06-0152          375.0    male  Glioblastoma Multiforme   
4  06-0157           97.0  female  Glioblastoma Multiforme   

                          center_name  
0  Broad Institute of MIT and Harvard  
1  Broad Institute of MIT and Harvard  
2  Broad Institute of MIT and Harvard  
3  Broad Institute of MIT and Harvard  
4  Broad Institute of MIT and Harvard

In [10]:
list(mt1)

['subject', 'days_to_death', 'gender1', 'disease_type', 'center_name']

# Read Metadata 2

In [11]:
mt2=pd.read_csv("TCGA/TCGA.Glioma.metadata.tsv",sep="\t")

In [12]:
mt2.shape

(1114, 18)

In [13]:
mt2.head()

case_submitter_id project_id primary_diagnosis   race  \
0      TCGA-02-0001   TCGA-GBM      Glioblastoma  white   
1      TCGA-02-0003   TCGA-GBM      Glioblastoma  white   
2      TCGA-02-0004   TCGA-GBM      Glioblastoma  white   
3      TCGA-02-0006   TCGA-GBM      Glioblastoma  white   
4      TCGA-02-0007   TCGA-GBM      Glioblastoma  white   

                ethnicity  gender RE_names IDH     x1p19q TERT  \
0  not hispanic or latino  female      NaN  WT  non-codel  NaN   
1  not hispanic or latino    male      NaN  WT  non-codel  NaN   
2  not hispanic or latino    male      NaN  WT        NaN  NaN   
3  not hispanic or latino  female      NaN  WT  non-codel  NaN   
4  not hispanic or latino  female      NaN  WT  non-codel  NaN   

         IDH_1p19q_status                           WHO_groups  \
0  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   
1  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   
2         IDH-WT:1p19q-NA                                  NaN   
3  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   
4  IDH-WT:1p19q-non-codel  glioblastoma:IDH-WT:1p19q-non-codel   

                     Triple_group     Tissue_sample_location          MGMT  \
0  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   
1  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   
2         IDH-WT:1p19q-NA:TERT-NA  MD Anderson Cancer Center           NaN   
3  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   
4  IDH-WT:1p19q-non-codel:TERT-NA  MD Anderson Cancer Center  Unmethylated   

    Age  Survival_months  Vital_status  
0  44.0        11.762054           1.0  
1  50.0         4.731106           1.0  
2  59.0        11.334941           1.0  
3  56.0        18.333034           1.0  
4  40.0        23.162705           1.0

In [14]:
list(mt2)

['case_submitter_id',
 'project_id',
 'primary_diagnosis',
 'race',
 'ethnicity',
 'gender',
 'RE_names',
 'IDH',
 'x1p19q',
 'TERT',
 'IDH_1p19q_status',
 'WHO_groups',
 'Triple_group',
 'Tissue_sample_location',
 'MGMT',
 'Age',
 'Survival_months',
 'Vital_status']

In [15]:
mt2['subject']=mt2['case_submitter_id'].astype(str).str[5:12]

In [16]:
mt2[['subject','case_submitter_id']].head()

subject case_submitter_id
0  02-0001      TCGA-02-0001
1  02-0003      TCGA-02-0003
2  02-0004      TCGA-02-0004
3  02-0006      TCGA-02-0006
4  02-0007      TCGA-02-0007

In [17]:
mt2.drop('case_submitter_id',axis='columns',inplace=True)
mt2.rename(columns={"gender": "gender2"},inplace=True)

In [18]:
list(mt2)

['project_id',
 'primary_diagnosis',
 'race',
 'ethnicity',
 'gender2',
 'RE_names',
 'IDH',
 'x1p19q',
 'TERT',
 'IDH_1p19q_status',
 'WHO_groups',
 'Triple_group',
 'Tissue_sample_location',
 'MGMT',
 'Age',
 'Survival_months',
 'Vital_status',
 'subject']

# Merge Metadata

In [19]:
#meta=mt2.merge(mt1,left_on='subject2',right_on='subject1',suffixes=["_mt2","_mt1"],how='inner')
meta=mt2.merge(mt1,on='subject',suffixes=["_mt2","_mt1"],how='inner')

In [20]:
del mt1
del mt2

In [21]:
meta.head()

project_id primary_diagnosis                       race  \
0   TCGA-GBM      Glioblastoma                      asian   
1   TCGA-GBM      Glioblastoma  black or african american   
2   TCGA-GBM      Glioblastoma                      white   
3   TCGA-GBM      Glioblastoma                      white   
4   TCGA-GBM      Glioblastoma                      white   

                ethnicity gender2                   RE_names     IDH  \
0  not hispanic or latino    male  02-2483-01A-01R-1849-01+1  Mutant   
1  not hispanic or latino    male  02-2485-01A-01R-1849-01+1      WT   
2  not hispanic or latino  female  06-0125-01A-01R-1849-01+1      WT   
3  not hispanic or latino    male                        NaN      WT   
4  not hispanic or latino  female  06-0157-01A-01R-1849-01+1      WT   

      x1p19q    TERT            IDH_1p19q_status  ...  \
0  non-codel      WT  IDH-Mutant:1p19q-non-codel  ...   
1  non-codel  Mutant      IDH-WT:1p19q-non-codel  ...   
2  non-codel  Mutant      IDH-WT:1p19q-non-codel  ...   
3  non-codel     NaN      IDH-WT:1p19q-non-codel  ...   
4  non-codel  Mutant      IDH-WT:1p19q-non-codel  ...   

      Tissue_sample_location          MGMT   Age Survival_months  \
0  MD Anderson Cancer Center    Methylated  43.0       15.310383   
1  MD Anderson Cancer Center  Unmethylated  53.0       15.441803   
2        Henry Ford Hospital    Methylated  63.0       47.573895   
3        Henry Ford Hospital           NaN  68.0       12.320588   
4        Henry Ford Hospital           NaN  63.0        3.186925   

   Vital_status  subject  days_to_death gender1             disease_type  \
0           0.0  02-2483            NaN    male  Glioblastoma Multiforme   
1           0.0  02-2485            NaN    male  Glioblastoma Multiforme   
2           1.0  06-0125         1448.0  female  Glioblastoma Multiforme   
3           1.0  06-0152          375.0    male  Glioblastoma Multiforme   
4           1.0  06-0157           97.0  female  Glioblastoma Multiforme   

                          center_name  
0  Broad Institute of MIT and Harvard  
1  Broad Institute of MIT and Harvard  
2  Broad Institute of MIT and Harvard  
3  Broad Institute of MIT and Harvard  
4  Broad Institute of MIT and Harvard  

[5 rows x 22 columns]

In [22]:
meta.drop(meta[(meta['subject']=='CS-6186') & (meta['center_name']=='Harvard Medical School')].index,inplace=True)

In [23]:
meta.drop(meta[(meta['subject']=='DU-5872') & (meta['center_name']=='Harvard Medical School')].index,inplace=True)

In [24]:
meta.columns

Index(['project_id', 'primary_diagnosis', 'race', 'ethnicity', 'gender2',
       'RE_names', 'IDH', 'x1p19q', 'TERT', 'IDH_1p19q_status', 'WHO_groups',
       'Triple_group', 'Tissue_sample_location', 'MGMT', 'Age',
       'Survival_months', 'Vital_status', 'subject', 'days_to_death',
       'gender1', 'disease_type', 'center_name'],
      dtype='object')

# need a multi-level index to merge with df later

In [25]:
##meta=meta.append(pd.Series(name='blankrowforindex',dtype=object))
##meta.loc['blankrowforindex']=''

In [26]:
##meta.columns=[meta.columns,meta.loc['blankrowforindex']]
#meta.columns=[meta.columns,[''] * len(meta.columns)]

In [27]:
meta.columns

Index(['project_id', 'primary_diagnosis', 'race', 'ethnicity', 'gender2',
       'RE_names', 'IDH', 'x1p19q', 'TERT', 'IDH_1p19q_status', 'WHO_groups',
       'Triple_group', 'Tissue_sample_location', 'MGMT', 'Age',
       'Survival_months', 'Vital_status', 'subject', 'days_to_death',
       'gender1', 'disease_type', 'center_name'],
      dtype='object')

In [28]:
#meta.tail()

In [29]:
##meta.drop('blankrowforindex',axis='index',inplace=True)

In [30]:
##meta.tail()

In [31]:
#meta.columns.names=['','']

In [32]:
meta.columns

Index(['project_id', 'primary_diagnosis', 'race', 'ethnicity', 'gender2',
       'RE_names', 'IDH', 'x1p19q', 'TERT', 'IDH_1p19q_status', 'WHO_groups',
       'Triple_group', 'Tissue_sample_location', 'MGMT', 'Age',
       'Survival_months', 'Vital_status', 'subject', 'days_to_death',
       'gender1', 'disease_type', 'center_name'],
      dtype='object')

# Read raw data

In [33]:
#df=pd.read_csv("focused.csv.gz",header=[0,1,2,3],index_col=[0,1])
#df.columns.names=['sample','pup','hkle','q']
#df=df.T
df=pd.read_csv("focused.T.csv.gz",header=[0,1],index_col=[0,1,2,3])

df.reset_index(inplace=True)
df.head()


chromosome       sample     pup      hkle mapq    chr1                  \
position                                       1048000 1190000 1308000   
0           02-2483-01A  paired  HERVK113  Q20       0       0       0   
1           02-2483-01A  paired     SVA_A  Q20       0       0       0   
2           02-2483-01A  paired     SVA_B  Q20       0       0       0   
3           02-2483-01A  paired     SVA_C  Q20       0       0       0   
4           02-2483-01A  paired     SVA_D  Q20       0       0       0   

chromosome                          ...     chrY                             \
position   1409000 1410000 1580000  ... 19535000 19536000 19699000 19700000   
0                4       1       4  ...        0        0        2        0   
1                0       0       0  ...        0        0        0        0   
2                0       0       0  ...        0        0        0        0   
3                0       0       0  ...        0        0        0        0   
4                0       0       0  ...        0        0        0        0   

chromosome                                                        
position   20552000 20553000 20976000 21047000 21217000 21639000  
0                 0        0        0        0        0        0  
1                 0        0        0        0        0        0  
2                 0        0        0        0        0        0  
3                 0        0        0        0        0        0  
4                 0        0        0        0        0        0  

[5 rows x 17413 columns]

In [34]:
df['subject']=df['sample'].str[0:7]
df['subject']

0       02-2483
1       02-2483
2       02-2483
3       02-2483
4       02-2483
         ...   
1913    TQ-A8XE
1914    TQ-A8XE
1915    TQ-A8XE
1916    TQ-A8XE
1917    TQ-A8XE
Name: subject, Length: 1918, dtype: object

In [35]:
df.columns

MultiIndex([( 'sample',         ''),
            (    'pup',         ''),
            (   'hkle',         ''),
            (   'mapq',         ''),
            (   'chr1',  '1048000'),
            (   'chr1',  '1190000'),
            (   'chr1',  '1308000'),
            (   'chr1',  '1409000'),
            (   'chr1',  '1410000'),
            (   'chr1',  '1580000'),
            ...
            (   'chrY', '19536000'),
            (   'chrY', '19699000'),
            (   'chrY', '19700000'),
            (   'chrY', '20552000'),
            (   'chrY', '20553000'),
            (   'chrY', '20976000'),
            (   'chrY', '21047000'),
            (   'chrY', '21217000'),
            (   'chrY', '21639000'),
            ('subject',         '')],
           names=['chromosome', 'position'], length=17414)

# Do I really want to merge the meta data? No. Not really. Not yet.

In [36]:
#df=df.merge(meta,on='subject')

In [37]:
df.head()

chromosome       sample     pup      hkle mapq    chr1                  \
position                                       1048000 1190000 1308000   
0           02-2483-01A  paired  HERVK113  Q20       0       0       0   
1           02-2483-01A  paired     SVA_A  Q20       0       0       0   
2           02-2483-01A  paired     SVA_B  Q20       0       0       0   
3           02-2483-01A  paired     SVA_C  Q20       0       0       0   
4           02-2483-01A  paired     SVA_D  Q20       0       0       0   

chromosome                          ...     chrY                             \
position   1409000 1410000 1580000  ... 19536000 19699000 19700000 20552000   
0                4       1       4  ...        0        2        0        0   
1                0       0       0  ...        0        0        0        0   
2                0       0       0  ...        0        0        0        0   
3                0       0       0  ...        0        0        0        0   
4                0       0       0  ...        0        0        0        0   

chromosome                                               subject  
position   20553000 20976000 21047000 21217000 21639000           
0                 0        0        0        0        0  02-2483  
1                 0        0        0        0        0  02-2483  
2                 0        0        0        0        0  02-2483  
3                 0        0        0        0        0  02-2483  
4                 0        0        0        0        0  02-2483  

[5 rows x 17414 columns]

In [38]:
df.set_index(['subject','sample','pup','hkle','mapq'],inplace=True)

In [39]:
df.head()

chromosome                                  chr1                          \
position                                 1048000 1190000 1308000 1409000   
subject sample      pup    hkle     mapq                                   
02-2483 02-2483-01A paired HERVK113 Q20        0       0       0       4   
                           SVA_A    Q20        0       0       0       0   
                           SVA_B    Q20        0       0       0       0   
                           SVA_C    Q20        0       0       0       0   
                           SVA_D    Q20        0       0       0       0   

chromosome                                                                \
position                                 1410000 1580000 1581000 2211000   
subject sample      pup    hkle     mapq                                   
02-2483 02-2483-01A paired HERVK113 Q20        1       4       0       0   
                           SVA_A    Q20        0       0       0       0   
                           SVA_B    Q20        0       0       0       0   
                           SVA_C    Q20        0       0       0       0   
                           SVA_D    Q20        0       0       0       0   

chromosome                                                ...     chrY  \
position                                 2513000 2581000  ... 19535000   
subject sample      pup    hkle     mapq                  ...            
02-2483 02-2483-01A paired HERVK113 Q20        0       0  ...        0   
                           SVA_A    Q20        0       0  ...        0   
                           SVA_B    Q20        0       0  ...        0   
                           SVA_C    Q20        0       0  ...        0   
                           SVA_D    Q20        0       0  ...        0   

chromosome                                                                    \
position                                 19536000 19699000 19700000 20552000   
subject sample      pup    hkle     mapq                                       
02-2483 02-2483-01A paired HERVK113 Q20         0        2        0        0   
                           SVA_A    Q20         0        0        0        0   
                           SVA_B    Q20         0        0        0        0   
                           SVA_C    Q20         0        0        0        0   
                           SVA_D    Q20         0        0        0        0   

chromosome                                                                    \
position                                 20553000 20976000 21047000 21217000   
subject sample      pup    hkle     mapq                                       
02-2483 02-2483-01A paired HERVK113 Q20         0        0        0        0   
                           SVA_A    Q20         0        0        0        0   
                           SVA_B    Q20         0        0        0        0   
                           SVA_C    Q20         0        0        0        0   
                           SVA_D    Q20         0        0        0        0   

chromosome                                         
position                                 21639000  
subject sample      pup    hkle     mapq           
02-2483 02-2483-01A paired HERVK113 Q20         0  
                           SVA_A    Q20         0  
                           SVA_B    Q20         0  
                           SVA_C    Q20         0  
                           SVA_D    Q20         0  

[5 rows x 17409 columns]

In [40]:
df.shape

(1918, 17409)

# Remove "pup" and "q" as they are useless in this dataset.

In [41]:
#df.columns.droplevel(1)
df=df.droplevel('pup',axis='index').droplevel('mapq',axis='index')

In [42]:
df.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19535000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19536000 19699000 19700000 20552000 20553000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        0        2        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                                        
position                     20976000 21047000 21217000 21639000  
subject sample      hkle                                          
02-2483 02-2483-01A HERVK113        0        0        0        0  
                    SVA_A           0        0        0        0  
                    SVA_B           0        0        0        0  
                    SVA_C           0        0        0        0  
                    SVA_D           0        0        0        0  

[5 rows x 17409 columns]

In [43]:
df.shape

(1918, 17409)

# Drop recurrent tumor (-02?)

In [44]:
df.loc[df.index.get_level_values('sample').str.contains("-02.$")].index

MultiIndex([('06-0125', '06-0125-02A', 'HERVK113'),
            ('06-0125', '06-0125-02A',    'SVA_A'),
            ('06-0125', '06-0125-02A',    'SVA_B'),
            ('06-0125', '06-0125-02A',    'SVA_C'),
            ('06-0125', '06-0125-02A',    'SVA_D'),
            ('06-0125', '06-0125-02A',    'SVA_E'),
            ('06-0125', '06-0125-02A',    'SVA_F'),
            ('06-0152', '06-0152-02A', 'HERVK113'),
            ('06-0152', '06-0152-02A',    'SVA_A'),
            ('06-0152', '06-0152-02A',    'SVA_B'),
            ...
            ('TQ-A7RV', 'TQ-A7RV-02A',    'SVA_D'),
            ('TQ-A7RV', 'TQ-A7RV-02A',    'SVA_E'),
            ('TQ-A7RV', 'TQ-A7RV-02A',    'SVA_F'),
            ('TQ-A8XE', 'TQ-A8XE-02A', 'HERVK113'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_A'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_B'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_C'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_D'),
            ('TQ-A8XE', 'TQ-A8XE-02A',    'SVA_E

In [45]:
df.drop(df.loc[df.index.get_level_values('sample').str.contains("-02.$")].index,axis='index',inplace=True)

In [46]:
df.loc[df.index.get_level_values('sample').str.contains("-02.$")].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

# Drop solid normal (-11?) (I've seen a couple)

In [47]:
df.loc[df.index.get_level_values('sample').str.contains("-11.$")].index

MultiIndex([('FG-5963', 'FG-5963-11A', 'HERVK113'),
            ('FG-5963', 'FG-5963-11A',    'SVA_A'),
            ('FG-5963', 'FG-5963-11A',    'SVA_B'),
            ('FG-5963', 'FG-5963-11A',    'SVA_C'),
            ('FG-5963', 'FG-5963-11A',    'SVA_D'),
            ('FG-5963', 'FG-5963-11A',    'SVA_E'),
            ('FG-5963', 'FG-5963-11A',    'SVA_F')],
           names=['subject', 'sample', 'hkle'])

In [48]:
df.drop(df.loc[df.index.get_level_values('sample').str.contains("-11.$")].index,axis='index',inplace=True)

In [49]:
df.loc[df.index.get_level_values('sample').str.contains("-11.$")].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

# Drop solitary tumor / normal ( not sure how to do this programmatically )

* FG-5963-01A
* CS-5390-10A

In [50]:
df.loc[df.index.get_level_values('sample') == 'FG-5963-01A'].index

MultiIndex([('FG-5963', 'FG-5963-01A', 'HERVK113'),
            ('FG-5963', 'FG-5963-01A',    'SVA_A'),
            ('FG-5963', 'FG-5963-01A',    'SVA_B'),
            ('FG-5963', 'FG-5963-01A',    'SVA_C'),
            ('FG-5963', 'FG-5963-01A',    'SVA_D'),
            ('FG-5963', 'FG-5963-01A',    'SVA_E'),
            ('FG-5963', 'FG-5963-01A',    'SVA_F')],
           names=['subject', 'sample', 'hkle'])

In [51]:
df.drop(df.loc[df.index.get_level_values('sample') == 'FG-5963-01A'].index,axis='index',inplace=True)

In [52]:
df.loc[df.index.get_level_values('sample') == 'FG-5963-01A'].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

In [53]:
df.loc[df.index.get_level_values('sample') == "CS-5390-10A"].index

MultiIndex([('CS-5390', 'CS-5390-10A', 'HERVK113'),
            ('CS-5390', 'CS-5390-10A',    'SVA_A'),
            ('CS-5390', 'CS-5390-10A',    'SVA_B'),
            ('CS-5390', 'CS-5390-10A',    'SVA_C'),
            ('CS-5390', 'CS-5390-10A',    'SVA_D'),
            ('CS-5390', 'CS-5390-10A',    'SVA_E'),
            ('CS-5390', 'CS-5390-10A',    'SVA_F')],
           names=['subject', 'sample', 'hkle'])

In [54]:
df.drop(df.loc[df.index.get_level_values('sample') == "CS-5390-10A"].index,axis='index',inplace=True)

In [55]:
df.loc[df.index.get_level_values('sample') == "CS-5390-10A"].index

MultiIndex([], names=['subject', 'sample', 'hkle'])

# Get raw read counts to normalize

In [56]:
c=pd.read_csv("TCGA/TCGA_raw_counts.csv")
c.rename(columns={"subject": "sample"},inplace=True)
#c.set_index('sample',inplace=True)
c.head()

sample  raw pair count
0  02-2483-01A       566024474
1  02-2483-10A       579304076
2  02-2485-01A       575060540
3  02-2485-10A       478851061
4  06-0125-01A      1066268796

In [57]:
#c.loc['CS-6665-01A','raw pair count']

In [58]:
#c.loc['CS-6665-10A','raw pair count']

In [59]:
#df.head()

In [60]:
#c.columns=[c.columns,['''',]]
c.columns=[c.columns,[''] * len(c.columns)]

In [61]:
c.head()

sample raw pair count
                             
0  02-2483-01A      566024474
1  02-2483-10A      579304076
2  02-2485-01A      575060540
3  02-2485-10A      478851061
4  06-0125-01A     1066268796

In [62]:
df.reset_index(inplace=True)

In [63]:
df=df.merge(c,on='sample')

/Users/jakewendt/Library/Python/3.6/lib/python/site-packages/pandas/core/generic.py:3889: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.
  obj = obj._drop_axis(labels, axis, level=level, errors=errors)


Not sure if this has any impact.

/Users/jakewendt/Library/Python/3.6/lib/python/site-packages/pandas/core/generic.py:3889: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.
  obj = obj._drop_axis(labels, axis, level=level, errors=errors)


In [64]:
df.head()

chromosome  subject       sample      hkle    chr1                          \
position                                   1048000 1190000 1308000 1409000   
0           02-2483  02-2483-01A  HERVK113       0       0       0       4   
1           02-2483  02-2483-01A     SVA_A       0       0       0       0   
2           02-2483  02-2483-01A     SVA_B       0       0       0       0   
3           02-2483  02-2483-01A     SVA_C       0       0       0       0   
4           02-2483  02-2483-01A     SVA_D       0       0       0       0   

chromosome                          ...     chrY                             \
position   1410000 1580000 1581000  ... 19536000 19699000 19700000 20552000   
0                1       4       0  ...        0        2        0        0   
1                0       0       0  ...        0        0        0        0   
2                0       0       0  ...        0        0        0        0   
3                0       0       0  ...        0        0        0        0   
4                0       0       0  ...        0        0        0        0   

chromosome                                              raw pair count  
position   20553000 20976000 21047000 21217000 21639000                 
0                 0        0        0        0        0      566024474  
1                 0        0        0        0        0      566024474  
2                 0        0        0        0        0      566024474  
3                 0        0        0        0        0      566024474  
4                 0        0        0        0        0      566024474  

[5 rows x 17413 columns]

In [65]:
df.set_index(['subject','sample','hkle'],inplace=True)
#df.set_index(['subject','sample','hkle','raw pair count'],inplace=True)

In [66]:
df.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19536000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19699000 19700000 20552000 20553000 20976000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        2        0        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                              raw pair count  
position                     21047000 21217000 21639000                 
subject sample      hkle                                                
02-2483 02-2483-01A HERVK113        0        0        0      566024474  
                    SVA_A           0        0        0      566024474  
                    SVA_B           0        0        0      566024474  
                    SVA_C           0        0        0      566024474  
                    SVA_D           0        0        0      566024474  

[5 rows x 17410 columns]

# Normalize (not sure if necessary here)

In [67]:
#dfn=df.copy()

In [68]:
#dfn.shape

In [69]:
df.columns

MultiIndex([(          'chr1',  '1048000'),
            (          'chr1',  '1190000'),
            (          'chr1',  '1308000'),
            (          'chr1',  '1409000'),
            (          'chr1',  '1410000'),
            (          'chr1',  '1580000'),
            (          'chr1',  '1581000'),
            (          'chr1',  '2211000'),
            (          'chr1',  '2513000'),
            (          'chr1',  '2581000'),
            ...
            (          'chrY', '19536000'),
            (          'chrY', '19699000'),
            (          'chrY', '19700000'),
            (          'chrY', '20552000'),
            (          'chrY', '20553000'),
            (          'chrY', '20976000'),
            (          'chrY', '21047000'),
            (          'chrY', '21217000'),
            (          'chrY', '21639000'),
            ('raw pair count',         '')],
           names=['chromosome', 'position'], length=17410)

In [70]:
chromosome_columns=df.columns[df.columns.get_level_values('chromosome').str.contains("^chr")]
chromosome_columns

MultiIndex([('chr1',  '1048000'),
            ('chr1',  '1190000'),
            ('chr1',  '1308000'),
            ('chr1',  '1409000'),
            ('chr1',  '1410000'),
            ('chr1',  '1580000'),
            ('chr1',  '1581000'),
            ('chr1',  '2211000'),
            ('chr1',  '2513000'),
            ('chr1',  '2581000'),
            ...
            ('chrY', '19535000'),
            ('chrY', '19536000'),
            ('chrY', '19699000'),
            ('chrY', '19700000'),
            ('chrY', '20552000'),
            ('chrY', '20553000'),
            ('chrY', '20976000'),
            ('chrY', '21047000'),
            ('chrY', '21217000'),
            ('chrY', '21639000')],
           names=['chromosome', 'position'], length=17409)

In [71]:
df.loc['CS-6665','CS-6665-01A']

chromosome    chr1                                                          \
position   1048000 1190000 1308000 1409000 1410000 1580000 1581000 2211000   
hkle                                                                         
HERVK113         0       0       0       6       3       2       0       0   
SVA_A            0       0       0       0       0       0       0       0   
SVA_B            0       0       0       0       1       0       1       0   
SVA_C            0       0       0       0       0       0       0       0   
SVA_D            0       0       0       0       0       0       0       0   
SVA_E            0       0       0       0       0       0       0       0   
SVA_F            0       0       0       0       0       0       0       0   

chromosome                  ...     chrY                                      \
position   2513000 2581000  ... 19536000 19699000 19700000 20552000 20553000   
hkle                        ...                                                
HERVK113         0       0  ...        0        0        0        0        0   
SVA_A            0       0  ...        0        0        0        0        0   
SVA_B            0       0  ...        0        0        0        0        0   
SVA_C            0       0  ...        0        0        0        0        0   
SVA_D            0       0  ...        0        0        0        0        0   
SVA_E            0       0  ...        0        0        0        0        0   
SVA_F            0       0  ...        0        0        0        0        0   

chromosome                                     raw pair count  
position   20976000 21047000 21217000 21639000                 
hkle                                                           
HERVK113          0        0        0        0      673209306  
SVA_A             0        0        0        0      673209306  
SVA_B             0        0        0        0      673209306  
SVA_C             0        0        0        0      673209306  
SVA_D             0        0        0        0      673209306  
SVA_E             0        0        0        0      673209306  
SVA_F             0        0        0        0      673209306  

[7 rows x 17410 columns]

In [72]:
df.loc['CS-6665','CS-6665-10A']

chromosome    chr1                                                          \
position   1048000 1190000 1308000 1409000 1410000 1580000 1581000 2211000   
hkle                                                                         
HERVK113         0       0       0       3       5       9       0       0   
SVA_A            0       0       0       0       0       0       0       0   
SVA_B            0       0       0       0       1       0       1       0   
SVA_C            0       0       0       0       0       0       0       0   
SVA_D            0       0       0       0       0       0       0       0   
SVA_E            0       0       0       0       0       0       0       0   
SVA_F            0       0       0       0       0       0       0       0   

chromosome                  ...     chrY                                      \
position   2513000 2581000  ... 19536000 19699000 19700000 20552000 20553000   
hkle                        ...                                                
HERVK113         0       0  ...        0        0        0        0        0   
SVA_A            0       0  ...        0        0        0        0        0   
SVA_B            0       0  ...        0        0        0        0        0   
SVA_C            0       0  ...        0        0        0        0        0   
SVA_D            0       0  ...        0        0        0        0        0   
SVA_E            0       0  ...        0        0        0        0        0   
SVA_F            0       0  ...        0        0        0        0        0   

chromosome                                     raw pair count  
position   20976000 21047000 21217000 21639000                 
hkle                                                           
HERVK113          0        0        0        0      343696933  
SVA_A             0        0        0        0      343696933  
SVA_B             0        0        0        0      343696933  
SVA_C             0        0        0        0      343696933  
SVA_D             0        0        0        0      343696933  
SVA_E             0        0        0        0      343696933  
SVA_F             0        0        0        0      343696933  

[7 rows x 17410 columns]

In [73]:
#df[chromosome_columns]=(1e9*df[chromosome_columns].values)/df[['raw pair count']].values

In [74]:
df.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19536000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19699000 19700000 20552000 20553000 20976000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        2        0        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                              raw pair count  
position                     21047000 21217000 21639000                 
subject sample      hkle                                                
02-2483 02-2483-01A HERVK113        0        0        0      566024474  
                    SVA_A           0        0        0      566024474  
                    SVA_B           0        0        0      566024474  
                    SVA_C           0        0        0      566024474  
                    SVA_D           0        0        0      566024474  

[5 rows x 17410 columns]

In [75]:
df.drop("raw pair count",axis='columns',inplace=True)

In [76]:
df.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19535000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19536000 19699000 19700000 20552000 20553000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        0        2        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                                        
position                     20976000 21047000 21217000 21639000  
subject sample      hkle                                          
02-2483 02-2483-01A HERVK113        0        0        0        0  
                    SVA_A           0        0        0        0  
                    SVA_B           0        0        0        0  
                    SVA_C           0        0        0        0  
                    SVA_D           0        0        0        0  

[5 rows x 17409 columns]

In [77]:
df.shape

(1708, 17409)

In [78]:
df.loc['CS-6665','CS-6665-01A']

chromosome    chr1                                                          \
position   1048000 1190000 1308000 1409000 1410000 1580000 1581000 2211000   
hkle                                                                         
HERVK113         0       0       0       6       3       2       0       0   
SVA_A            0       0       0       0       0       0       0       0   
SVA_B            0       0       0       0       1       0       1       0   
SVA_C            0       0       0       0       0       0       0       0   
SVA_D            0       0       0       0       0       0       0       0   
SVA_E            0       0       0       0       0       0       0       0   
SVA_F            0       0       0       0       0       0       0       0   

chromosome                  ...     chrY                                      \
position   2513000 2581000  ... 19535000 19536000 19699000 19700000 20552000   
hkle                        ...                                                
HERVK113         0       0  ...        0        0        0        0        0   
SVA_A            0       0  ...        0        0        0        0        0   
SVA_B            0       0  ...        0        0        0        0        0   
SVA_C            0       0  ...        0        0        0        0        0   
SVA_D            0       0  ...        0        0        0        0        0   
SVA_E            0       0  ...        0        0        0        0        0   
SVA_F            0       0  ...        0        0        0        0        0   

chromosome                                               
position   20553000 20976000 21047000 21217000 21639000  
hkle                                                     
HERVK113          0        0        0        0        0  
SVA_A             0        0        0        0        0  
SVA_B             0        0        0        0        0  
SVA_C             0        0        0        0        0  
SVA_D             0        0        0        0        0  
SVA_E             0        0        0        0        0  
SVA_F             0        0        0        0        0  

[7 rows x 17409 columns]

In [79]:
df.loc['CS-6665','CS-6665-10A']

chromosome    chr1                                                          \
position   1048000 1190000 1308000 1409000 1410000 1580000 1581000 2211000   
hkle                                                                         
HERVK113         0       0       0       3       5       9       0       0   
SVA_A            0       0       0       0       0       0       0       0   
SVA_B            0       0       0       0       1       0       1       0   
SVA_C            0       0       0       0       0       0       0       0   
SVA_D            0       0       0       0       0       0       0       0   
SVA_E            0       0       0       0       0       0       0       0   
SVA_F            0       0       0       0       0       0       0       0   

chromosome                  ...     chrY                                      \
position   2513000 2581000  ... 19535000 19536000 19699000 19700000 20552000   
hkle                        ...                                                
HERVK113         0       0  ...        0        0        0        0        0   
SVA_A            0       0  ...        0        0        0        0        0   
SVA_B            0       0  ...        0        0        0        0        0   
SVA_C            0       0  ...        0        0        0        0        0   
SVA_D            0       0  ...        0        0        0        0        0   
SVA_E            0       0  ...        0        0        0        0        0   
SVA_F            0       0  ...        0        0        0        0        0   

chromosome                                               
position   20553000 20976000 21047000 21217000 21639000  
hkle                                                     
HERVK113          0        0        0        0        0  
SVA_A             0        0        0        0        0  
SVA_B             0        0        0        0        0  
SVA_C             0        0        0        0        0  
SVA_D             0        0        0        0        0  
SVA_E             0        0        0        0        0  
SVA_F             0        0        0        0        0  

[7 rows x 17409 columns]

# Match earlier notebook counts
* CS-5390-10A - 553
* FG-5963-01A - 68
* CS-6186-01A - 1100
* CS-6186-10A - 950
* DU-5872-01A - 1108
* DU-5872-10A - 776
* 02-2483-01A - 568
* 02-2483-10A - 557
* 02-2485-01A - 587
* 02-2485-10A - 498
* 06-0125-01A - 734

In [80]:
df[df[df.columns]>0].count(axis='columns')['CS-6186','CS-6186-01A']

hkle
HERVK113    1100
SVA_A       2224
SVA_B       2141
SVA_C       2174
SVA_D       2200
SVA_E       2159
SVA_F       1784
dtype: int64

In [81]:
df[df[df.columns]>0].count(axis='columns')['CS-6186','CS-6186-10A']

hkle
HERVK113     950
SVA_A       1962
SVA_B       1839
SVA_C       1845
SVA_D       1868
SVA_E       1827
SVA_F       1512
dtype: int64

In [82]:
df[df[df.columns]>0].count(axis='columns')['DU-5872','DU-5872-01A']

hkle
HERVK113    1108
SVA_A       2744
SVA_B       2640
SVA_C       2625
SVA_D       2644
SVA_E       2614
SVA_F       2285
dtype: int64

In [83]:
df[df[df.columns]>0].count(axis='columns')['DU-5872','DU-5872-10A']

hkle
HERVK113     776
SVA_A       2008
SVA_B       1904
SVA_C       1908
SVA_D       1926
SVA_E       1884
SVA_F       1547
dtype: int64

# Keep only Broad and drop Harvard Medical School

In [84]:
meta[meta['center_name'] == 'Broad Institute of MIT and Harvard']['subject'].unique()


array(['02-2483', '02-2485', '06-0125', '06-0152', '06-0157', '06-0171',
       '06-0190', '06-0210', '06-0211', '06-0214', '06-0221', '06-0686',
       '06-0744', '06-0745', '06-2557', '06-2570', '06-5411', '06-5415',
       '14-1034', '14-1402', '14-1823', '14-2554', '15-1444', '19-1389',
       '19-2620', '19-2624', '19-2629', '19-5960', '26-5132', '26-5135',
       '27-1831', '27-2523', '27-2528', '32-1970', '41-5651', 'CS-5390',
       'CS-5395', 'CS-6186', 'CS-6665', 'CS-6668', 'CS-6669', 'DB-5278',
       'DH-A669', 'DU-5870', 'DU-5872', 'DU-5874', 'DU-6397', 'DU-6401',
       'DU-6404', 'DU-6407', 'DU-7009', 'DU-7013', 'DU-7301', 'DU-7304',
       'E1-5318', 'E1-5319', 'EZ-7264', 'FG-5964', 'FG-5965', 'FG-7643',
       'FG-8182', 'FG-A4MT', 'HT-7602', 'HT-7689', 'HT-7695', 'HT-8104',
       'HT-A5R7', 'HT-A61B', 'HW-7486', 'HW-7487', 'IK-7675', 'TM-A7CF',
       'TQ-A7RK', 'TQ-A7RV', 'TQ-A8XE'], dtype=object)

In [85]:
df.index.get_level_values('subject')

Index(['02-2483', '02-2483', '02-2483', '02-2483', '02-2483', '02-2483',
       '02-2483', '02-2483', '02-2483', '02-2483',
       ...
       'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE',
       'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE', 'TQ-A8XE'],
      dtype='object', name='subject', length=1708)

In [86]:
df.index.get_level_values('subject').isin(
    meta[meta['center_name'] == 'Broad Institute of MIT and Harvard']['subject'].unique())


array([ True,  True,  True, ...,  True,  True,  True])

In [87]:
df=df[df.index.get_level_values('subject').isin(
    meta[meta['center_name'] == 'Broad Institute of MIT and Harvard']['subject'].unique())]
df.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
                    SVA_A          0       0       0       0       0       0   
                    SVA_B          0       0       0       0       0       0   
                    SVA_C          0       0       0       0       0       0   
                    SVA_D          0       0       0       0       0       0   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19535000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
                    SVA_A          0       0       0       0  ...        0   
                    SVA_B          0       0       0       0  ...        0   
                    SVA_C          0       0       0       0  ...        0   
                    SVA_D          0       0       0       0  ...        0   

chromosome                                                                 \
position                     19536000 19699000 19700000 20552000 20553000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        0        2        0        0        0   
                    SVA_A           0        0        0        0        0   
                    SVA_B           0        0        0        0        0   
                    SVA_C           0        0        0        0        0   
                    SVA_D           0        0        0        0        0   

chromosome                                                        
position                     20976000 21047000 21217000 21639000  
subject sample      hkle                                          
02-2483 02-2483-01A HERVK113        0        0        0        0  
                    SVA_A           0        0        0        0  
                    SVA_B           0        0        0        0  
                    SVA_C           0        0        0        0  
                    SVA_D           0        0        0        0  

[5 rows x 17409 columns]

In [88]:
df.shape

(1022, 17409)

#  Separate individual HERVs and SVAs

In [89]:
#hervk113=df.loc[pd.IndexSlice[:,:,'HERVK113'],:]
#sva_a=df.loc[pd.IndexSlice[:,:,'SVA_A'],:]
#sva_b=df.loc[pd.IndexSlice[:,:,'SVA_B'],:]
#sva_c=df.loc[pd.IndexSlice[:,:,'SVA_C'],:]
#sva_d=df.loc[pd.IndexSlice[:,:,'SVA_D'],:]
#sva_e=df.loc[pd.IndexSlice[:,:,'SVA_E'],:]
#sva_f=df.loc[pd.IndexSlice[:,:,'SVA_F'],:]

hervk113=df.loc[df.index.get_level_values('hkle') == "HERVK113",:]
sva_a=df.loc[df.index.get_level_values('hkle') == 'SVA_A',:]
sva_b=df.loc[df.index.get_level_values('hkle') == 'SVA_B',:]
sva_c=df.loc[df.index.get_level_values('hkle') == 'SVA_C',:]
sva_d=df.loc[df.index.get_level_values('hkle') == 'SVA_D',:]
sva_e=df.loc[df.index.get_level_values('hkle') == 'SVA_E',:]
sva_f=df.loc[df.index.get_level_values('hkle') == 'SVA_F',:]

In [90]:
hervk113.head()

chromosome                      chr1                                          \
position                     1048000 1190000 1308000 1409000 1410000 1580000   
subject sample      hkle                                                       
02-2483 02-2483-01A HERVK113       0       0       0       4       1       4   
        02-2483-10A HERVK113       0       0       0       2       3       3   
02-2485 02-2485-01A HERVK113       0       0       0       3       4       4   
        02-2485-10A HERVK113       0       0       0       8       0       0   
06-0125 06-0125-01A HERVK113       0       0       0       9       6       9   

chromosome                                                    ...     chrY  \
position                     1581000 2211000 2513000 2581000  ... 19535000   
subject sample      hkle                                      ...            
02-2483 02-2483-01A HERVK113       0       0       0       0  ...        0   
        02-2483-10A HERVK113       0       0       0       0  ...        0   
02-2485 02-2485-01A HERVK113       0       0       0       0  ...        0   
        02-2485-10A HERVK113       0       0       0       0  ...        0   
06-0125 06-0125-01A HERVK113       0       0       0       0  ...        0   

chromosome                                                                 \
position                     19536000 19699000 19700000 20552000 20553000   
subject sample      hkle                                                    
02-2483 02-2483-01A HERVK113        0        2        0        0        0   
        02-2483-10A HERVK113        0        1        1        0        0   
02-2485 02-2485-01A HERVK113        0        0        1        0        0   
        02-2485-10A HERVK113        0        1        0        0        0   
06-0125 06-0125-01A HERVK113        0        0        0        0        0   

chromosome                                                        
position                     20976000 21047000 21217000 21639000  
subject sample      hkle                                          
02-2483 02-2483-01A HERVK113        0        0        0        0  
        02-2483-10A HERVK113        0        0        0        0  
02-2485 02-2485-01A HERVK113        1        0        0        0  
        02-2485-10A HERVK113        0        0        0        0  
06-0125 06-0125-01A HERVK113        0        0        0        0  

[5 rows x 17409 columns]

In [91]:
hervk113.shape

(146, 17409)

# Now the HKLE index is useless

In [92]:
#df=df.droplevel('hkle',axis='columns')

In [93]:
#df.head()

In [94]:
#df.shape

In [95]:
hervk113=hervk113.droplevel('hkle',axis='index')
sva_a=sva_a.droplevel('hkle',axis='index')
sva_b=sva_b.droplevel('hkle',axis='index')
sva_c=sva_c.droplevel('hkle',axis='index')
sva_d=sva_d.droplevel('hkle',axis='index')
sva_e=sva_e.droplevel('hkle',axis='index')
sva_f=sva_f.droplevel('hkle',axis='index')

#for ldf in hervk113,sva_a,sva_b,sva_c,sva_d,sva_e,sva_f:
#    ldf=ldf.droplevel('hkle',axis='index')


In [96]:
hervk113.head()

chromosome             chr1                                                  \
position            1048000 1190000 1308000 1409000 1410000 1580000 1581000   
subject sample                                                                
02-2483 02-2483-01A       0       0       0       4       1       4       0   
        02-2483-10A       0       0       0       2       3       3       0   
02-2485 02-2485-01A       0       0       0       3       4       4       0   
        02-2485-10A       0       0       0       8       0       0       0   
06-0125 06-0125-01A       0       0       0       9       6       9       0   

chromosome                                   ...     chrY                    \
position            2211000 2513000 2581000  ... 19535000 19536000 19699000   
subject sample                               ...                              
02-2483 02-2483-01A       0       0       0  ...        0        0        2   
        02-2483-10A       0       0       0  ...        0        0        1   
02-2485 02-2485-01A       0       0       0  ...        0        0        0   
        02-2485-10A       0       0       0  ...        0        0        1   
06-0125 06-0125-01A       0       0       0  ...        0        0        0   

chromosome                                                                 \
position            19700000 20552000 20553000 20976000 21047000 21217000   
subject sample                                                              
02-2483 02-2483-01A        0        0        0        0        0        0   
        02-2483-10A        1        0        0        0        0        0   
02-2485 02-2485-01A        1        0        0        1        0        0   
        02-2485-10A        0        0        0        0        0        0   
06-0125 06-0125-01A        0        0        0        0        0        0   

chromosome                    
position            21639000  
subject sample                
02-2483 02-2483-01A        0  
        02-2483-10A        0  
02-2485 02-2485-01A        0  
        02-2485-10A        0  
06-0125 06-0125-01A        0  

[5 rows x 17409 columns]

In [97]:
hervk113.shape

(146, 17409)

# Drop all those with 0 insertions

In [98]:
#for ldf in hervk113,sva_a,sva_b,sva_c,sva_d,sva_e,sva_f:
#    ldf.drop(ldf.loc[:,ldf.sum(axis='index')==0].columns,axis='columns',inplace=True)
    
hervk113.drop(hervk113.loc[:,hervk113.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_a.drop(sva_a.loc[:,sva_a.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_b.drop(sva_b.loc[:,sva_b.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_c.drop(sva_c.loc[:,sva_c.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_d.drop(sva_d.loc[:,sva_d.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_e.drop(sva_d.loc[:,sva_e.sum(axis='index')==0].columns,axis='columns',inplace=True)
sva_f.drop(sva_f.loc[:,sva_f.sum(axis='index')==0].columns,axis='columns',inplace=True)


In [99]:
hervk113.head()

chromosome             chr1                                                  \
position            1409000 1410000 1580000 3751000 5573000 6304000 8502000   
subject sample                                                                
02-2483 02-2483-01A       4       1       4       0       0       0       0   
        02-2483-10A       2       3       3       0       0       0       0   
02-2485 02-2485-01A       3       4       4       0       0       0       0   
        02-2485-10A       8       0       0       0       0       0       0   
06-0125 06-0125-01A       9       6       9       0       0       0       0   

chromosome                                   ...     chrY                    \
position            8912000 9001000 9002000  ... 12462000 12463000 12879000   
subject sample                               ...                              
02-2483 02-2483-01A       0       0       0  ...        1        1        0   
        02-2483-10A       0       0       1  ...        0        2        0   
02-2485 02-2485-01A       0       0       0  ...        1        0        0   
        02-2485-10A       0       0       0  ...        0        1        0   
06-0125 06-0125-01A       0       0       0  ...        0        0        0   

chromosome                                                                 \
position            13103000 13104000 15774000 19699000 19700000 20552000   
subject sample                                                              
02-2483 02-2483-01A        0        0        0        2        0        0   
        02-2483-10A        0        0        1        1        1        0   
02-2485 02-2485-01A        0        1        0        0        1        0   
        02-2485-10A        1        5        0        1        0        0   
06-0125 06-0125-01A        0        0        0        0        0        0   

chromosome                    
position            20976000  
subject sample                
02-2483 02-2483-01A        0  
        02-2483-10A        0  
02-2485 02-2485-01A        1  
        02-2485-10A        0  
06-0125 06-0125-01A        0  

[5 rows x 2645 columns]

In [100]:
hervk113.shape

(146, 2645)

In [101]:
sva_a.shape

(146, 5625)

In [102]:
sva_b.shape

(146, 5820)

In [103]:
sva_c.shape

(146, 11182)

In [104]:
sva_d.shape

(146, 11229)

In [105]:
sva_d.shape

(146, 11229)

In [106]:
sva_f.shape

(146, 5377)

# How to compare the precise site presence in tumor / normal pairs?

In [125]:
h=hervk113.copy()
h.head()

chromosome             chr1                                                  \
position            1409000 1410000 1580000 3751000 5573000 6304000 8502000   
subject sample                                                                
02-2483 02-2483-01A       4       1       4       0       0       0       0   
        02-2483-10A       2       3       3       0       0       0       0   
02-2485 02-2485-01A       3       4       4       0       0       0       0   
        02-2485-10A       8       0       0       0       0       0       0   
06-0125 06-0125-01A       9       6       9       0       0       0       0   

chromosome                                   ...     chrY                    \
position            8912000 9001000 9002000  ... 12462000 12463000 12879000   
subject sample                               ...                              
02-2483 02-2483-01A       0       0       0  ...        1        1        0   
        02-2483-10A       0       0       1  ...        0        2        0   
02-2485 02-2485-01A       0       0       0  ...        1        0        0   
        02-2485-10A       0       0       0  ...        0        1        0   
06-0125 06-0125-01A       0       0       0  ...        0        0        0   

chromosome                                                                 \
position            13103000 13104000 15774000 19699000 19700000 20552000   
subject sample                                                              
02-2483 02-2483-01A        0        0        0        2        0        0   
        02-2483-10A        0        0        1        1        1        0   
02-2485 02-2485-01A        0        1        0        0        1        0   
        02-2485-10A        1        5        0        1        0        0   
06-0125 06-0125-01A        0        0        0        0        0        0   

chromosome                    
position            20976000  
subject sample                
02-2483 02-2483-01A        0  
        02-2483-10A        0  
02-2485 02-2485-01A        1  
        02-2485-10A        0  
06-0125 06-0125-01A        0  

[5 rows x 2645 columns]

In [128]:
for subject in hervk113.index.get_level_values('subject'):
    h.loc[subject]=[
        (((
            hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[0]>0)),
        (((
            hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[1]>0))]

In [129]:
h.head()

chromosome             chr1                                                  \
position            1409000 1410000 1580000 3751000 5573000 6304000 8502000   
subject sample                                                                
02-2483 02-2483-01A   False   False   False   False   False   False   False   
        02-2483-10A   False   False   False   False   False   False   False   
02-2485 02-2485-01A   False    True    True   False   False   False   False   
        02-2485-10A   False   False   False   False   False   False   False   
06-0125 06-0125-01A   False   False   False   False   False   False   False   

chromosome                                   ...     chrY                    \
position            8912000 9001000 9002000  ... 12462000 12463000 12879000   
subject sample                               ...                              
02-2483 02-2483-01A   False   False   False  ...     True    False    False   
        02-2483-10A   False   False    True  ...    False    False    False   
02-2485 02-2485-01A   False   False   False  ...     True    False    False   
        02-2485-10A   False   False   False  ...    False     True    False   
06-0125 06-0125-01A   False   False   False  ...    False    False    False   

chromosome                                                                 \
position            13103000 13104000 15774000 19699000 19700000 20552000   
subject sample                                                              
02-2483 02-2483-01A    False    False    False    False    False    False   
        02-2483-10A    False    False     True    False     True    False   
02-2485 02-2485-01A    False    False    False    False     True    False   
        02-2485-10A     True    False    False     True    False    False   
06-0125 06-0125-01A    False    False    False    False    False    False   

chromosome                    
position            20976000  
subject sample                
02-2483 02-2483-01A    False  
        02-2483-10A    False  
02-2485 02-2485-01A     True  
        02-2485-10A    False  
06-0125 06-0125-01A    False  

[5 rows x 2645 columns]

In [158]:
hervk113[h].head()

chromosome             chr1                                                  \
position            1409000 1410000 1580000 3751000 5573000 6304000 8502000   
subject sample                                                                
02-2483 02-2483-01A     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
        02-2483-10A     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
02-2485 02-2485-01A     NaN     4.0     4.0     NaN     NaN     NaN     NaN   
        02-2485-10A     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
06-0125 06-0125-01A     NaN     NaN     NaN     NaN     NaN     NaN     NaN   

chromosome                                   ...     chrY                    \
position            8912000 9001000 9002000  ... 12462000 12463000 12879000   
subject sample                               ...                              
02-2483 02-2483-01A     NaN     NaN     NaN  ...      1.0      NaN      NaN   
        02-2483-10A     NaN     NaN     1.0  ...      NaN      NaN      NaN   
02-2485 02-2485-01A     NaN     NaN     NaN  ...      1.0      NaN      NaN   
        02-2485-10A     NaN     NaN     NaN  ...      NaN      1.0      NaN   
06-0125 06-0125-01A     NaN     NaN     NaN  ...      NaN      NaN      NaN   

chromosome                                                                 \
position            13103000 13104000 15774000 19699000 19700000 20552000   
subject sample                                                              
02-2483 02-2483-01A      NaN      NaN      NaN      NaN      NaN      NaN   
        02-2483-10A      NaN      NaN      1.0      NaN      1.0      NaN   
02-2485 02-2485-01A      NaN      NaN      NaN      NaN      1.0      NaN   
        02-2485-10A      1.0      NaN      NaN      1.0      NaN      NaN   
06-0125 06-0125-01A      NaN      NaN      NaN      NaN      NaN      NaN   

chromosome                    
position            20976000  
subject sample                
02-2483 02-2483-01A      NaN  
        02-2483-10A      NaN  
02-2485 02-2485-01A      1.0  
        02-2485-10A      NaN  
06-0125 06-0125-01A      NaN  

[5 rows x 2645 columns]

NaN means that neither tumor nor normal were unique at that position.

sum==0 mean that position is never unique.

In [150]:
hervk113.loc[:,hervk113[h].fillna(0).sum(axis='index')==0].head()

chromosome              chr1                       chr12    chr13            \
position            15379000 46242000 223404000 48552000 90090000 100761000   
subject sample                                                                
02-2483 02-2483-01A        8        0         0        0        0         0   
        02-2483-10A        5        0         0        0        0         0   
02-2485 02-2485-01A        8        0         0        0        0         0   
        02-2485-10A        3        0         0        0        0         0   
06-0125 06-0125-01A        7        0         0        0        0         0   

chromosome             chr19    chr22      chr3     chr4     chr5           \
position            21658000 11555000 175131000 68694000 65092000 75606000   
subject sample                                                               
02-2483 02-2483-01A        0        0         0        0        0        3   
        02-2483-10A        0        0         0        0        0        3   
02-2485 02-2485-01A        0        0         0        0        0        7   
        02-2485-10A        0        0         0        0        0        5   
06-0125 06-0125-01A        0        0         0        0        4       10   

chromosome                       chr6      chr7 chrUn_GL000219v1  
position            88616000 16004000 158980000           175000  
subject sample                                                    
02-2483 02-2483-01A        0        0         0               13  
        02-2483-10A        0        0         0               16  
02-2485 02-2485-01A        0        0         0               16  
        02-2485-10A        0        0         0                9  
06-0125 06-0125-01A        0        0         0               21

In [151]:
hervk113.loc[:,hervk113[h].fillna(0).sum(axis='index')>0].head()

chromosome             chr1                                                  \
position            1409000 1410000 1580000 3751000 5573000 6304000 8502000   
subject sample                                                                
02-2483 02-2483-01A       4       1       4       0       0       0       0   
        02-2483-10A       2       3       3       0       0       0       0   
02-2485 02-2485-01A       3       4       4       0       0       0       0   
        02-2485-10A       8       0       0       0       0       0       0   
06-0125 06-0125-01A       9       6       9       0       0       0       0   

chromosome                                   ...     chrY                    \
position            8912000 9001000 9002000  ... 12462000 12463000 12879000   
subject sample                               ...                              
02-2483 02-2483-01A       0       0       0  ...        1        1        0   
        02-2483-10A       0       0       1  ...        0        2        0   
02-2485 02-2485-01A       0       0       0  ...        1        0        0   
        02-2485-10A       0       0       0  ...        0        1        0   
06-0125 06-0125-01A       0       0       0  ...        0        0        0   

chromosome                                                                 \
position            13103000 13104000 15774000 19699000 19700000 20552000   
subject sample                                                              
02-2483 02-2483-01A        0        0        0        2        0        0   
        02-2483-10A        0        0        1        1        1        0   
02-2485 02-2485-01A        0        1        0        0        1        0   
        02-2485-10A        1        5        0        1        0        0   
06-0125 06-0125-01A        0        0        0        0        0        0   

chromosome                    
position            20976000  
subject sample                
02-2483 02-2483-01A        0  
        02-2483-10A        0  
02-2485 02-2485-01A        1  
        02-2485-10A        0  
06-0125 06-0125-01A        0  

[5 rows x 2629 columns]

In [165]:
hervk113.loc[:,hervk113[h].fillna(0).sum(axis='index')>0]

chromosome             chr1                                                  \
position            1409000 1410000 1580000 3751000 5573000 6304000 8502000   
subject sample                                                                
02-2483 02-2483-01A       4       1       4       0       0       0       0   
        02-2483-10A       2       3       3       0       0       0       0   
02-2485 02-2485-01A       3       4       4       0       0       0       0   
        02-2485-10A       8       0       0       0       0       0       0   
06-0125 06-0125-01A       9       6       9       0       0       0       0   
...                     ...     ...     ...     ...     ...     ...     ...   
TQ-A7RK TQ-A7RK-10A       5       0       5       0       0       0       0   
TQ-A7RV TQ-A7RV-01A       6       3       6       0       0       0       0   
        TQ-A7RV-10A       4       2       7       0       0       0       0   
TQ-A8XE TQ-A8XE-01A       5       5       2       0       1       1       0   
        TQ-A8XE-10A       6       4       6       1       2       0       0   

chromosome                                   ...     chrY                    \
position            8912000 9001000 9002000  ... 12462000 12463000 12879000   
subject sample                               ...                              
02-2483 02-2483-01A       0       0       0  ...        1        1        0   
        02-2483-10A       0       0       1  ...        0        2        0   
02-2485 02-2485-01A       0       0       0  ...        1        0        0   
        02-2485-10A       0       0       0  ...        0        1        0   
06-0125 06-0125-01A       0       0       0  ...        0        0        0   
...                     ...     ...     ...  ...      ...      ...      ...   
TQ-A7RK TQ-A7RK-10A       0       3       0  ...        0        1        0   
TQ-A7RV TQ-A7RV-01A       0       3       0  ...        1        3        0   
        TQ-A7RV-10A       1       1       1  ...        0        3        0   
TQ-A8XE TQ-A8XE-01A       0       2       1  ...        0        0        0   
        TQ-A8XE-10A       0       5       2  ...        0        0        0   

chromosome                                                                 \
position            13103000 13104000 15774000 19699000 19700000 20552000   
subject sample                                                              
02-2483 02-2483-01A        0        0        0        2        0        0   
        02-2483-10A        0        0        1        1        1        0   
02-2485 02-2485-01A        0        1        0        0        1        0   
        02-2485-10A        1        5        0        1        0        0   
06-0125 06-0125-01A        0        0        0        0        0        0   
...                      ...      ...      ...      ...      ...      ...   
TQ-A7RK TQ-A7RK-10A        0        3        2        0        2        0   
TQ-A7RV TQ-A7RV-01A        0        3        1        1        1        0   
        TQ-A7RV-10A        0        3        0        0        1        1   
TQ-A8XE TQ-A8XE-01A        0        0        0        0        0        0   
        TQ-A8XE-10A        0        0        0        0        0        0   

chromosome                    
position            20976000  
subject sample                
02-2483 02-2483-01A        0  
        02-2483-10A        0  
02-2485 02-2485-01A        1  
        02-2485-10A        0  
06-0125 06-0125-01A        0  
...                      ...  
TQ-A7RK TQ-A7RK-10A        0  
TQ-A7RV TQ-A7RV-01A        0  
        TQ-A7RV-10A        0  
TQ-A8XE TQ-A8XE-01A        0  
        TQ-A8XE-10A        0  

[146 rows x 2629 columns]

In [162]:
(hervk113[h]/hervk113[h]).fillna(0).sum(axis='index')==0

chromosome  position
chr1        1409000     False
            1410000     False
            1580000     False
            3751000     False
            5573000     False
                        ...  
chrY        15774000    False
            19699000    False
            19700000    False
            20552000    False
            20976000    False
Length: 2645, dtype: bool

In [167]:
hervk113.index.get_level_values('subject')

Index(['02-2483', '02-2483', '02-2485', '02-2485', '06-0125', '06-0125',
       '06-0157', '06-0157', '06-0171', '06-0171',
       ...
       'IK-7675', 'IK-7675', 'TM-A7CF', 'TM-A7CF', 'TQ-A7RK', 'TQ-A7RK',
       'TQ-A7RV', 'TQ-A7RV', 'TQ-A8XE', 'TQ-A8XE'],
      dtype='object', name='subject', length=146)

In [166]:
hervk113.columns

MultiIndex([('chr1',  '1409000'),
            ('chr1',  '1410000'),
            ('chr1',  '1580000'),
            ('chr1',  '3751000'),
            ('chr1',  '5573000'),
            ('chr1',  '6304000'),
            ('chr1',  '8502000'),
            ('chr1',  '8912000'),
            ('chr1',  '9001000'),
            ('chr1',  '9002000'),
            ...
            ('chrY', '12462000'),
            ('chrY', '12463000'),
            ('chrY', '12879000'),
            ('chrY', '13103000'),
            ('chrY', '13104000'),
            ('chrY', '15774000'),
            ('chrY', '19699000'),
            ('chrY', '19700000'),
            ('chrY', '20552000'),
            ('chrY', '20976000')],
           names=['chromosome', 'position'], length=2645)

In [169]:
htest=pd.DataFrame(columns=hervk113.columns,index=hervk113.index.get_level_values('subject'))

In [178]:
for subject in hervk113.index.get_level_values('subject'):
    tm=(((
            hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[0]>0))
    t=hervk113.loc[subject].iloc[0][tm] #.fillna(0)
    print(t)
    nm=(((
            hervk113.loc[subject].iloc[0]>0)^(
            hervk113.loc[subject].iloc[1]>0))&(
            hervk113.loc[subject].iloc[1]>0))
    n=hervk113.loc[subject].iloc[1][nm] #.fillna(0)
    print(n)
    #print(t-n)

#    htest.loc[subject]=t-n


chromosome  position 
chr1        11891000     1
            15135000     1
            46320000     4
            54637000     1
            65140000     1
                        ..
chrX        138333000    1
            145720000    2
            145721000    1
            149627000    1
chrY        12462000     1
Name: 02-2483-01A, Length: 180, dtype: int64
chromosome  position 
chr1        9002000      1
            29338000     1
            39234000     1
            46395000     1
            59915000     1
                        ..
chrX        135406000    1
chrY        6749000      2
            7180000      2
            15774000     1
            19700000     1
Name: 02-2483-10A, Length: 169, dtype: int64
chromosome  position 
chr1        11891000     1
            15135000     1
            46320000     4
            54637000     1
            65140000     1
                        ..
chrX        138333000    1
            145720000    2
            145721000    1
       

chromosome  position 
chr1        12784000     1
            15135000     2
            27854000     1
            29360000     2
            46320000     2
                        ..
chrX        145721000    1
chrY        6749000      4
            7180000      1
            12463000     1
            20976000     1
Name: 06-0214-10A, Length: 179, dtype: int64
chromosome  position 
chr1        9002000      1
            10425000     3
            10427000     5
            11891000     1
            13114000     1
                        ..
chrX        139342000    1
            152130000    1
chrY        6748000      2
            7711000      1
            13103000     2
Name: 06-0221-01A, Length: 246, dtype: int64
chromosome  position 
chr1        1580000      4
            93277000     2
            150632000    1
            160652000    1
            237127000    1
                        ..
chrX        1899000      1
            57336000     1
            90951000     2
       

chromosome  position 
chr1        10425000     4
            10427000     1
            12784000     1
            13353000     1
            15135000     1
                        ..
chrX        123679000    1
            142323000    1
            152130000    1
            154522000    1
            154909000    1
Name: 14-1402-01A, Length: 228, dtype: int64
chromosome  position 
chr1        29338000     1
            148138000    1
            156180000    1
            156181000    4
            160949000    1
                        ..
chrX        23659000     1
            48555000     1
            108376000    1
            135406000    2
            138296000    1
Name: 14-1402-10A, Length: 90, dtype: int64
chromosome  position 
chr1        10427000     4
            21389000     1
            29338000     1
            46395000     1
            54637000     3
                        ..
chrX        26525000     1
            48555000     1
            77931000     1
        

chromosome  position 
chr1        10425000     2
            15136000     5
            27220000     1
            29338000     1
            46396000     1
                        ..
chrX        152129000    1
chrY        7180000      1
            7711000      1
            13103000     1
            15774000     1
Name: 27-1831-01A, Length: 161, dtype: int64
chromosome  position 
chr1        1409000      1
            10427000     3
            33068000     1
            73129000     1
            108518000    2
                        ..
chr9        121429000    1
            122447000    1
            131356000    1
chrX        87973000     1
chrY        3782000      1
Name: 27-1831-10A, Length: 148, dtype: int64
chromosome  position 
chr1        10425000     2
            15136000     5
            27220000     1
            29338000     1
            46396000     1
                        ..
chrX        152129000    1
chrY        7180000      1
            7711000      1
       

chromosome  position 
chr1        1410000      4
            9002000      3
            26819000     1
            45529000     1
            54636000     1
                        ..
chrX        57336000     2
            108376000    1
            123679000    2
            138296000    2
            145721000    1
Name: CS-6669-01A, Length: 218, dtype: int64
chromosome  position 
chr1        12784000     1
            15136000     3
            39234000     1
            46395000     3
            104135000    1
                        ..
chrX        1292000      2
            1899000      1
            55998000     2
            87973000     3
            113991000    1
Name: CS-6669-10A, Length: 98, dtype: int64
chromosome  position 
chr1        10425000     2
            13353000     1
            29359000     1
            65139000     1
            73129000     1
                        ..
chrX        125780000    1
            138333000    1
            145720000    1
        

chromosome  position 
chr1        3751000      1
            9001000      5
            33068000     1
            34948000     1
            46321000     1
                        ..
chrX        145721000    1
            155907000    1
chrY        7180000      1
            7711000      2
            12463000     3
Name: DU-7013-01A, Length: 409, dtype: int64
chromosome  position 
chr1        10425000     2
            25588000     1
            45081000     1
            45248000     1
            52182000     1
                        ..
chrX        149627000    2
chrY        7179000      1
            12879000     1
            13103000     2
            19699000     2
Name: DU-7013-10A, Length: 163, dtype: int64
chromosome  position 
chr1        3751000      1
            9001000      5
            33068000     1
            34948000     1
            46321000     1
                        ..
chrX        145721000    1
            155907000    1
chrY        7180000      1
       

chromosome  position 
chr1        15135000     2
            39234000     2
            40572000     8
            45528000     5
            45529000     1
                        ..
chrX        145721000    1
            152129000    2
            152130000    2
chrY        6749000      1
            7179000      2
Name: FG-8182-01A, Length: 200, dtype: int64
chromosome  position 
chr1        70461000     1
            147106000    1
            152455000    2
            166605000    1
            213150000    1
                        ..
chr9        133958000    2
chrX        48555000     1
            72733000     1
            145720000    1
chrY        19700000     3
Name: FG-8182-10A, Length: 81, dtype: int64
chromosome  position 
chr1        9001000      4
            21054000     2
            26803000     1
            33063000     1
            54637000     3
                        ..
chrX        100724000    1
            113990000    1
            135030000    1
        

chromosome  position 
chr1        9002000      1
            40000000     1
            108517000    2
            108518000    2
            111800000    1
                        ..
chrX        55998000     1
            90950000     2
            138333000    2
chrY        7179000      1
            19700000     1
Name: HW-7487-01A, Length: 151, dtype: int64
chromosome  position 
chr1        1409000      3
            15135000     1
            29359000     2
            45528000     1
            46395000     1
                        ..
chrX        154909000    1
chrY        7711000      1
            12463000     2
            13103000     1
            19699000     2
Name: HW-7487-10A, Length: 166, dtype: int64
chromosome  position 
chr1        5573000      1
            10427000     2
            11891000     1
            15135000     1
            27854000     1
                        ..
chrX        152129000    5
chrY        7180000      1
            12462000     1
       

In [171]:
htest

chromosome    chr1                                                          \
position   1409000 1410000 1580000 3751000 5573000 6304000 8502000 8912000   
subject                                                                      
02-2483        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
02-2483        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
02-2485        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
02-2485        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
06-0125        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
...            ...     ...     ...     ...     ...     ...     ...     ...   
TQ-A7RK        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
TQ-A7RV        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
TQ-A7RV        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
TQ-A8XE        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
TQ-A8XE        NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   

chromosome                  ...     chrY                                      \
position   9001000 9002000  ... 12462000 12463000 12879000 13103000 13104000   
subject                     ...                                                
02-2483        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
02-2483        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
02-2485        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
02-2485        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
06-0125        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
...            ...     ...  ...      ...      ...      ...      ...      ...   
TQ-A7RK        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
TQ-A7RV        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
TQ-A7RV        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
TQ-A8XE        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   
TQ-A8XE        NaN     NaN  ...      NaN      NaN      NaN      NaN      NaN   

chromosome                                               
position   15774000 19699000 19700000 20552000 20976000  
subject                                                  
02-2483         NaN      NaN      NaN      NaN      NaN  
02-2483         NaN      NaN      NaN      NaN      NaN  
02-2485         NaN      NaN      NaN      NaN      NaN  
02-2485         NaN      NaN      NaN      NaN      NaN  
06-0125         NaN      NaN      NaN      NaN      NaN  
...             ...      ...      ...      ...      ...  
TQ-A7RK         NaN      NaN      NaN      NaN      NaN  
TQ-A7RV         NaN      NaN      NaN      NaN      NaN  
TQ-A7RV         NaN      NaN      NaN      NaN      NaN  
TQ-A8XE         NaN      NaN      NaN      NaN      NaN  
TQ-A8XE         NaN      NaN      NaN      NaN      NaN  

[146 rows x 2645 columns]

In [153]:
stop before here

SyntaxError: invalid syntax (<ipython-input-153-5d2a6af1dcd5>, line 1)

In [108]:
##dftns.loc['TQ-A8XE',0:,:]>0

##dftns.loc[idx['TQ-A8XE',1,:]]
##dftns.loc['TQ-A8XE'].index.get_level_values(0)[0]

#dftns.loc['TQ-A8XE',:,:]>0
dftns.loc['TQ-A8XE',:]>0

NameError: name 'dftns' is not defined

In [ ]:
#dftns.loc['TQ-A8XE','TQ-A8XE-01A',:]>0
dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0

In [ ]:
#dftns.loc['TQ-A8XE','TQ-A8XE-10A',:]>0
dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0

# OR

In [ ]:
#(dftns.loc['TQ-A8XE','TQ-A8XE-01A',724443745]>0)|(dftns.loc['TQ-A8XE','TQ-A8XE-10A',:]>0)

(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)|(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


# AND

In [ ]:
(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)&(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


# A XOR B ( means only A or B, never both )

In [ ]:
(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


# ( A XOR B ) AND A ( means only A )

# ( A XOR B ) AND B ( means only B )


In [ ]:
((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)


In [ ]:
((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)


In [ ]:
(((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)).sum()


In [ ]:
(((dftns.loc['TQ-A8XE','TQ-A8XE-01A']>0)^(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0))&(dftns.loc['TQ-A8XE','TQ-A8XE-10A']>0)).sum()


In [ ]:
dftns.loc['TQ-A8XE'].iloc[0]

In [ ]:
dftns.loc['TQ-A8XE'].iloc[1]

In [ ]:
dftns.index.levels[0]

In [ ]:
dftns.index.get_level_values(0)

In [ ]:
(dftns.loc['TQ-A8XE'].iloc[0]>0).sum()

In [ ]:
#np.set_printoptions(threshold=np.inf)

In [ ]:
dftns.loc['TQ-A8XE'].iloc[0].values

In [ ]:
#np.set_printoptions(threshold=1000)

In [ ]:
max_rows=pd.get_option('display.max_rows')

In [ ]:
#pd.set_option('display.max_rows', None)

In [ ]:
dftns.head()

In [ ]:
#pd.DataFrame(df[(df['diff']>10) | (df['diff']<-10)].index)

In [ ]:
pd.set_option('display.max_rows', max_rows)

In [ ]:
stn=pd.DataFrame(columns=['subject','uniq tumor','uniq normal'])
stn.set_index('subject',inplace=True)

sstu=pd.DataFrame(columns=['subject','sample','tn','uniq'])
sstu.set_index(['subject','sample','tn'],inplace=True)


In [ ]:
for i in dftns.index.levels[0]:
    stn.loc[i]=((((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[0]>0)).sum(),
               (((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[1]>0)).sum())
    sstu.loc[(i,dftns.loc[i].iloc[0].name, 'tumor'),:]=(
        ((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[0]>0)).sum()
    sstu.loc[(i,dftns.loc[i].iloc[1].name,'normal'),:]=(
        ((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[1]>0)).sum()

# Plot tries to parse the subject ids into dates because they are ##-####
# I can't figure out how to stop it.
#    abctn.loc[("_"+i,dftns.loc[i].iloc[0].name, 'tumor'),:]=(
#((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[0]>0)).sum()
#    abctn.loc[("_"+i,dftns.loc[i].iloc[1].name,'normal'),:]=(
#((dftns.loc[i].iloc[0]>0)^(dftns.loc[i].iloc[1]>0))&(dftns.loc[i].iloc[1]>0)).sum()
    

In [ ]:
dftns.loc['02-2483'].iloc[1].name

In [ ]:
stn

In [ ]:
sstu

In [ ]:
stn.plot(kind='bar',y=['uniq normal','uniq tumor'])
plt.xticks(rotation=75,ha='right')
plt.show()

In [ ]:
sstu.head()

In [ ]:
sstu.reset_index(inplace=True)

In [ ]:
sstu.head()

In [ ]:
groups = sstu.groupby('tn')
fig, ax = plt.subplots()
plt.xticks(rotation=90)
for name, group in groups:
    ax.bar(group['sample'],group['uniq'], label=name)
#    ax.bar(group['subject'],group['uniq'], label=name)
ax.legend()
plt.show()

In [ ]:
meta.head()

In [ ]:
sstum=sstu.merge(meta,on='subject')

In [ ]:
sstum.head()

In [ ]:
groups = sstum.groupby('disease_type')
fig, ax = plt.subplots()
plt.xticks(rotation=90)
for name, group in groups:
    ax.bar(group['sample'],group['uniq'], label=name)
ax.legend()
plt.show()

In [ ]:
sstum.boxplot(column='uniq',
    by=['disease_type','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['IDH','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['TERT','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['x1p19q','tn'])

In [ ]:
sstum.boxplot(column='uniq',
    by=['disease_type','tn','IDH','TERT','x1p19q'])
plt.xticks(rotation=75,ha='right')
plt.show()

In [ ]:
sstum.set_index(['subject','sample','tn'],inplace=True)

In [ ]:
sstum.head()

# Simple sample / tumor count / normal count / disease type plots

In [ ]:
stnm=stn.merge(meta,left_index=True,right_on='subject')

In [ ]:
stnm.set_index('subject',inplace=True)

In [ ]:
stnm.index

In [ ]:
stnm.head()

In [ ]:
stnm.groupby('disease_type').plot(kind='bar',y=['uniq normal','uniq tumor'])


# END Simple sample / tumor count / normal count / disease type plots

In [ ]:
#sstu.reset_index(inplace=True)
#sstu.head()

In [ ]:
#sstum=sstu.merge(meta,on='subject')

In [ ]:
#sstum.head()

# Transpose again

In [ ]:
dftnt=dftn.T

In [ ]:
dftnt.head()

# Number of sites for each subject

In [ ]:
dftnt['normal_count']=dftnt.loc[:,
    dftnt.columns.str.match('.*-10.$')][ dftnt.loc[:,dftnt.columns.str.match('.*-10.$')]>1 ].count(axis=1)

In [ ]:
dftnt['normal_count'].plot(kind='bar')

In [ ]:
dftnt['tumor_count']=dftnt.loc[:,
    dftnt.columns.str.match('.*-01.$')][ dftnt.loc[:,dftnt.columns.str.match('.*-01.$')]>1 ].count(axis=1)

In [ ]:
dftnt['tumor_count'].plot(kind='bar')

# Diff is normal - tumor. Negative means more in tumor.

In [ ]:
dftnt['diff']=dftnt['normal_count']-dftnt['tumor_count']

In [ ]:
dftnt['diff']

In [ ]:
dftnt['diff'].plot(kind='bar')

In [ ]:
dftnt[(dftnt['diff']>5) | (dftnt['diff']<-5)]['diff'].plot(kind='bar')

In [ ]:
dftnt[(dftnt['diff']>10) | (dftnt['diff']<-10)]['diff'].plot(kind='bar')
plt.xticks(rotation=75,ha='right')
plt.show()

In [ ]:
initial_max_rows=pd.get_option('display.max_rows')

In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
pd.DataFrame(index=dftnt[(dftnt['diff']>10) | (dftnt['diff']<-10)].index,
             data=dftnt[['diff','normal_count','tumor_count']])


In [ ]:
pd.set_option('display.max_rows', initial_max_rows)

In [ ]:
#df.loc[:,df.columns.str.match('.*-01.$')].shape

In [ ]:
#tumor=df.loc[:,df.columns.str.match('.*-01.$')]

In [ ]:
#tumor.shape

In [ ]:
#df.loc[:,df.columns.str.match('.*-10.$')].shape

In [ ]:
#normal=df.loc[:,df.columns.str.match('.*-10.$')]

In [ ]:
#normal.shape

In [ ]:
#normal[normal>0].count(axis=1)

In [ ]:
#normal.loc[:,normal.loc['chr1',1409000]>0]

In [ ]:
#tumor.loc[:,tumor.loc['chr1',1409000]>0]

In [ ]:
#len(normal.loc[:,normal.loc['chr1',1409000]>0].columns)

In [ ]:
#df['normal_count']=df.loc[:,
#    df.columns.str.match('.*-10.$')][ df.loc[:,df.columns.str.match('.*-10.$')]>0 ].count(axis=1)

In [ ]:
#df['tumor_count']=df.loc[:,
#    df.columns.str.match('.*-01.$')][ df.loc[:,df.columns.str.match('.*-01.$')]>0 ].count(axis=1)

In [ ]:
#df['diff']=df['normal_count']-df['tumor_count']

In [ ]:
#df['diff']

In [ ]:
#df['diff'].plot(kind='bar')

In [ ]:
#df[(df['diff']>10) | (df['diff']<-10)]['diff'].plot(kind='bar')

In [ ]:
max_rows=pd.get_option('display.max_rows')

In [ ]:
pd.set_option('display.max_rows', None)

In [ ]:
#pd.DataFrame(df[(df['diff']>10) | (df['diff']<-10)].index)

In [ ]:
pd.set_option('display.max_rows', max_rows)